### PR TITLE
Attempt to diagnose WinHTTP CI pipeline errors.

### LINKED ISSUE #5154: WinHttp Transport logging (and exception strings) have dangling periods making it harder to read due to new line in errorMsg from winhttp
`errorString.get(); ` returns something like "Foo\n" and hence adding the trailing period (and subsequent DWORD error to string) ends up in a new line.

From https://github.com/Azure/azure-sdk-for-cpp/blob/083d52aefccdaa721a96cc9b9537f727126f8b70/sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp#L401-L425

The logs, which also end up emitting the exception message (in the transport layer) end up like this:
```text
[2023-11-10T23:26:43.1726294Z T: 38480] INFO  : Status operation: 2097152(WINHTTP_CALLBACK_STATUS_REQUEST_ERROR )
[2023-11-10T23:26:43.1733686Z T: 38480] ERROR : Request error.  Error Code: 12002: The operation timed out
. 1
[2023-11-10T23:26:43.1738532Z T: 38480] DEBUG : WinHttpRequest::~WinHttpRequest. Closing handle synchronously.
```
image

cc @LarryOsterman

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

Closes #5154

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]
👍 
Closes #5154?

File: sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp [hunk 1]
```diff
@@ -421,6 +421,15 @@ std::string GetErrorMessage(DWORD error)
...
+    // If the end of the error message is a CRLF, remove it.
+    if (errorMessage.back() == '\n')
+    {
+      errorMessage.erase(errorMessage.size() - 1);
+      if (errorMessage.back() == '\r')
+      {
+        errorMessage.erase(errorMessage.size() - 1);
+      }
+    }
   }
```
Comment (ahsonkhan) [inline sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp:433]: Nice!

Do we still see a trailing/dangling "error code" after the period? I am not sure where that's being emitted from.

From the issue, note that extra ". 1":
> [2023-11-10T23:26:43.1733686Z T: 38480] ERROR : Request error.  Error Code: 12002: The operation timed out
. 1

I am assuming it is somewhere in this `GetErrorAndThrow()` call which has a defaulted `GetLastError()` param, but the ". 1" seems to come after this method returns the `errorMessage`:
https://github.com/Azure/azure-sdk-for-cpp/blob/dffa3edb2af2ab2673d0baad85cf4f49b7be9713/sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp#L1227-L1246

Comment (ahsonkhan) [reply sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp:433]: Actually, it is not the response log message, but rather the request one.

Its this one, within `OnHttpStatusOperation()`, before this PR:
https://github.com/Azure/azure-sdk-for-cpp/blob/2337ee796e4ef8a48cb1d25a45a581a3c8b7bffb/sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp#L597-L602

Which this PR is fixing (line 641-642). So, we should be in a more readable state now :)

**Pull request merged**